osi_environment.proto

```diff
@@ -324,4 +336,163 @@ message EnvironmentalConditions
         //
         optional uint32 seconds_since_midnight = 1;
     }
+
+    //
+    // \brief Specification of a cloud layer.
+    //
+    message CloudLayer
+    {
+        // Definition of the cloud layer, i.e. cloud layer level and sky visualization settings.
+        //
+        optional CloudLayerLevel cloud_layer_level = 1;
+
+        // Defines the cloud layer levels given by observation of total cloud amount in oktas.
+        //
+        // For visual alignment please see reference[1].
+        //
+        // The total degree of coverage indicates how large the part of the sky vault is which is covered
+        // with clouds altogether. It is given in eighths, because this division is easier to estimate
+        // for the observer. 0 eighths means that there are no traces of clouds in the sky,
```

Comment:
"because this division is easier to estimate for the observer." Can be removed; reasons for oktas need not to be stated here.